src/components/Web3Provider/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { Web3ReactProvider } from '@web3-react/core'
+import { Connector } from '@web3-react/types'
+import {
+  coinbaseWalletHooks,
+  createOrderedConnectors,
+  fortmaticHooks,
+  getConnectorForWallet,
+  gnosisSafe,
+  injectedHooks,
+  MODAL_WALLETS,
+  network,
+  Wallet,
+  walletConnectHooks,
+} from 'connectors'
+import usePrevious from 'hooks/usePrevious'
+import { useEffect, useState } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { updateWalletOverride } from 'state/walletOverride/reducer'
+
+const connect = async (connector: Connector) => {
+  try {
+    if (connector.connectEagerly) {
+      await connector.connectEagerly()
+    } else {
+      await connector.activate()
+    }
+  } catch (error) {
+    console.debug(`web3-react error: ${typeof connector}, ${error}`)
+  }
+}
+
+interface ConnectorState {
+  isActive: boolean
+  previousIsActive: boolean | undefined
+}
+
+// This component handles state changes in web3-react. It eagerly connects to all wallets.
+// It also checks for Coinbase Wallet, Wallet Connect Fortmatic or Injected wallets to become active.
+function Web3Updater() {
+  const dispatch = useAppDispatch()
+
+  const walletOverride = useAppSelector((state) => state.walletOverride.walletOverride)
+  const walletOverrideBackfilled = useAppSelector((state) => state.walletOverride.walletOverrideBackfilled)
+
+  const injectedIsActive = injectedHooks.useIsActive()
+  const previousInjectedIsActive = usePrevious(injectedIsActive)
+
+  const coinbaseWalletIsActive = coinbaseWalletHooks.useIsActive()
+  const previousCoinbaseWalletIsActive = usePrevious(coinbaseWalletIsActive)
+
+  const walletConnectIsActive = walletConnectHooks.useIsActive()
+  const previousWalletConnectIsActive = usePrevious(walletConnectIsActive)
+
+  const fortmaticIsActive = fortmaticHooks.useIsActive()
+  const previousFortmaticIsActive = usePrevious(fortmaticIsActive)
+
+  const [isEagerlyConnecting, setIsEagerlyConnecting] = useState(false)
+
+  // The dependency list is empty so this is only run once on mount
+  useEffect(() => {
+    connect(gnosisSafe)
+    connect(network)
+
+    if (walletOverride) {
```

Comment:
If you set the type of `walletOverride?: Wallet | true`, this will need to be more specific, maybe to `typeof walletOverride === 'string'`?